libbeat/processors/actions/decode_json_fields.go

```diff
@@ -75,7 +80,58 @@ func (f decodeJSONFields) Run(event common.MapStr) (common.MapStr, error) {
 				continue
 			}
 
-			_, err = event.Put(field, output)
+			if f.target != nil {
+				if len(*f.target) > 0 {
+					_, err = event.Put(*f.target, output)
+				} else {
+					switch t := output.(type) {
```

Comment:
What do you think of adding a global helper function for that switch part. Because now we duplicate code. For example, `Helpers.JsonHelper.OverwriteJsonKeys`?